python/cudf/cudf/core/column/lists.py

```diff
@@ -228,3 +229,50 @@ def len(self):
         dtype: int32
         """
         return self._return_or_inplace(count_elements(self._column))
+
+    def take(self, lists_indices):
+        """
+        Collect list elements based on given indices.
+
+        Parameters
+        ----------
+        lists_indices: List type arrays
+                       Specifies what to collect from each row
+
+        Returns
+        -------
+        ListColumn
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], None, [4, 5]])
+        >>> s
+        0    [1, 2, 3]
+        1         None
+        2       [4, 5]
+        dtype: list
+        >>> s.list.take([[0, 1], [], []])
+        0    [1, 2]
+        1      None
+        2        []
+        dtype: list
+        """
+
+        lists_indices_col = as_column(lists_indices)
+        if not isinstance(lists_indices_col, ListColumn):
+            raise ValueError("lists_indices should be list type array.")
```

Comment:
What's the behavior of `Series.take(...)` with a scalar input? I'm wondering if passing an Integer Column to here would make sense to map to `extract_list_element` under the hood or if that should be a separate API?

If it would make sense under the hood, we should probably raise a `NotImplementedError` if a NumericalColumn is passed in.

Reply:
To clarify, this is the expected behavior right?
```
s = [[1, 2, 3], [3], [4, 5]]
s.list.take([2, 0, 1]) --> s.list.extract_list_element([2, 0, 1]) --> [3, 3, 5]
```

Reply:
Yes, that's the expected behavior.

Reply:
Looks like `pd.Series.take` does not work with scalars, so I'm not sure if we want to overload `take` in this context to handle scalars.

Reply:
Seems like the scalar equivalent for `Series` is `Series.iat` which then has a `__getitem__` implementation, but I really don't like that.

Reply:
Got it. Currently, the planned `extract_list_element` only take a single scalar for input, is `NumericalColumn` planned to be added?

Reply:
Maybe I'm mixing things up in my head. Either way lets just continue with the PR as is as it's the correct behavior.